cmake/cppcheck.cmake

```diff
@@ -0,0 +1,9 @@
+find_program(CMAKE_CXX_CPPCHECK NAMES cppcheck)
+if(CMAKE_CXX_CPPCHECK)
+    list(APPEND CMAKE_CXX_CPPCHECK 
+        "--error-exitcode=1"
+        "--enable=warning,style"
+        "--force" 
+        "--inline-suppr"
+        "--suppressions-list=${CMAKE_SOURCE_DIR}/CppCheckSuppressions.txt")
```

Comment:
The input files are not specified, hence this does nothing.

Reply:
CMake automatically adds all source files here.

Reply:
I'm not sure about that. But it looks like cppcheck is not even invoked during the build since there is no output in the build logs and the build should fail, because there are still some open issues.

Reply:
Are you sure. Works for me. Did you enable it with `cmake .. -DCPR_ENABLE_CPPCHECK=ON`?

Reply:
And the CI run also has the output for it:
```bash
[...]
  Checking /home/runner/work/cpr/cpr/cpr/curlholder.cpp: CPR_ENABLE_CPPCHECK=1;CPR_ENABLE_SSL=1;cpr_EXPORTS=1;NDEBUG=1;__TINYC__...
  Checking /home/runner/work/cpr/cpr/cpr/curlholder.cpp: CPR_ENABLE_CPPCHECK=1;CPR_ENABLE_SSL=1;cpr_EXPORTS=1;NDEBUG=1;__TURBOC__...
  Checking /home/runner/work/cpr/cpr/cpr/curlholder.cpp: CPR_ENABLE_CPPCHECK=1;CPR_ENABLE_SSL=1;cpr_EXPORTS=1;NDEBUG=1;__VAX;__VMS...
  Checking /home/runner/work/cpr/cpr/cpr/curlholder.cpp: CPR_ENABLE_CPPCHECK=1;CPR_ENABLE_SSL=1;cpr_EXPORTS=1;NDEBUG=1;__VMS...
  Checking /home/runner/work/cpr/cpr/cpr/curlholder.cpp: CPR_ENABLE_CPPCHECK=1;CPR_ENABLE_SSL=1;cpr_EXPORTS=1;NDEBUG=1;__VXWORKS__;__WATCOMC__...
  Checking /home/runner/work/cpr/cpr/cpr/curlholder.cpp: CPR_ENABLE_CPPCHECK=1;CPR_ENABLE_SSL=1;cpr_EXPORTS=1;NDEBUG=1;__WATCOMC__...
  Checking /home/runner/work/cpr/cpr/cpr/curlholder.cpp: CPR_ENABLE_CPPCHECK=1;CPR_ENABLE_SSL=1;cpr_EXPORTS=1;NDEBUG=1;__xlc__...
[...]
```

Reply:
You're right. I got confused because the output looks different from what I expected. A disadvantage of the current approach is that cppcheck is invoked for each source file separately. Should you ever consider searching for unused functions, this will be a problem.